### PR TITLE
fix!: limit snappable features to current view extent

### PR DESCRIPTION
**Pull request by megawac**
# Purpose

- Avoids confusing guides being generated from features outside of the view. Users of our app at least found this confusing as they would not know what they were snapping to
- Can significantly improves performance on maps with many features as many features can be excluded by the r tree

I can make this an option if preferred.

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title is formatted as a [conventional-commit](https://www.conventionalcommits.org/) message.

### CODE REVIEW

File: src/control/cad.js [hunk 1]
```diff
@@ -251,7 +251,7 @@ class CadControl extends Control {
    */
   getClosestFeatures(coordinate, numFeatures) {
     const num = numFeatures || 1;
-    const ext = [-Infinity, -Infinity, Infinity, Infinity];
+    const ext = this.map.getView().calculateExtent();
```
Comment (oterral) [inline src/control/cad.js:254]: I can see different use case here. I propose to add a property which take a function. I do it master. Thanks for pointing this out.

**Review by oterral — COMMENTED** [review]
This PR is not relevant with the current codebase. We remove the use of `forEachFeatureInExtent` because it was useless when  useSpatialIndex is set to false on the source.
But I can easily imagine that can be a performance issue. I will add a property to be able to use also extent.

**Comment by oterral** [comment]
@megawac  I published 2.4.0 . Can you try the new extentFilter property and sees if it fits your need ?

**Comment by megawac** [comment]
Sounds promising. Will take a look when I am working on this area of our project again.

**Comment by oterral** [comment]
I close this,  feel free to reopen once you have tested the extentFilter property.

**Pull request closed without merging**